scheduled/cru-org-sync.js

```diff
@@ -0,0 +1,104 @@
+'use strict'
+
+const sequelize = require('../config/sequelize')
+const Recommendation = require('../models/recommendation')
+
+const request = require('request-promise-native')
+const {assign, find, forEach} = require('lodash')
+const prefixMatch = new RegExp('^https?://(www\\.)?cru\\.org(/content/cru)?', 'i')
+const scoresQuery = `SELECT MD5(lower(uri)) AS id, uri AS url, score FROM scores WHERE uri ~* $1`
+
+module.exports.handler = async (lambdaEvent) => {
+  /**
+   * Fetches cru.org json information through querybuilder
+   * @returns {Promise<void>}
+   */
+  const getCruOrgJson = async () => {
+    const hosts = ['prodpub1.aws.cru.org:4503', 'prodpub2.aws.cru.org:4503']
+    const options = {
+      method: 'GET',
+      qs: {
+        path: '/content/cru/',
+        'p.limit': '-1',
+        type: 'cq:Page',
+        'p.hits': 'selective',
+        'p.properties': 'jcr:path jcr:content/jcr:title jcr:content/jcr:description jcr:content/subtitle jcr:content/image/fileReference'
+      },
+      headers: {
+        'Cache-Control': 'no-cache',
+        'User-Agent': 'Request-Promise'
+      },
+      json: true
+    }
+
+    return new Promise((resolve, reject) => {
+      request(assign({}, options, {url: `http://${hosts[0]}/bin/querybuilder.json`})).then(resolve, () => {
+        request(assign({}, options, {url: `http://${hosts[1]}/bin/querybuilder.json`})).then(resolve, reject)
+      })
+    })
+  }
+
+  /**
+   * Combine scored url with cruOrg Json
+   * @param record
+   * @param cruOrgJson
+   * @returns {*}
+   */
+  const scoreTransform = (record, cruOrgJson) => {
+    let recommendation = new Recommendation(record)
```

Comment:
Does this just fill in the recommendation fields that score (record) contains?

Reply:
Yes, it sets `id`, `url` and `score` as returned by the scores query.